[WebGPU] Rename the setting from just "WebGPU" to "WebGPUEnabled"

litherum opened the pull request:
#### 529569b680163e9d37ed746acdf2f74d5035120f
<pre>
[WebGPU] Rename the setting from just &quot;WebGPU&quot; to &quot;WebGPUEnabled&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=256575">https://bugs.webkit.org/show_bug.cgi?id=256575</a>
rdar://109134933

Reviewed by Simon Fraser and Mike Wyrzykowski.

This is so we can say settings().webGPUEnabled() instead of settings().webGPU(),
which is a bit more ergonomic.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/WebGPU/GPU.idl:
* Source/WebCore/Modules/WebGPU/GPUAdapter.idl:
* Source/WebCore/Modules/WebGPU/GPUAdapterInfo.idl:
* Source/WebCore/Modules/WebGPU/GPUAddressMode.idl:
* Source/WebCore/Modules/WebGPU/GPUAutoLayoutMode.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroup.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupEntry.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupLayoutDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupLayoutEntry.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendComponent.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendFactor.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendOperation.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendState.idl:
* Source/WebCore/Modules/WebGPU/GPUBuffer.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferBinding.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferBindingType.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferMapState.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferUsage.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasCompositingAlphaMode.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl:
* Source/WebCore/Modules/WebGPU/GPUColorDict.idl:
* Source/WebCore/Modules/WebGPU/GPUColorTargetState.idl:
* Source/WebCore/Modules/WebGPU/GPUColorWrite.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandBuffer.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandBufferDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandEncoderDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandsMixin.idl:
* Source/WebCore/Modules/WebGPU/GPUCompareFunction.idl:
* Source/WebCore/Modules/WebGPU/GPUCompilationInfo.idl:
* Source/WebCore/Modules/WebGPU/GPUCompilationMessage.idl:
* Source/WebCore/Modules/WebGPU/GPUCompilationMessageType.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassTimestampLocation.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassTimestampWrite.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePipeline.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePipelineDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUCullMode.idl:
* Source/WebCore/Modules/WebGPU/GPUDebugCommandsMixin.idl:
* Source/WebCore/Modules/WebGPU/GPUDepthStencilState.idl:
* Source/WebCore/Modules/WebGPU/GPUDevice.idl:
* Source/WebCore/Modules/WebGPU/GPUDeviceDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUDeviceLostInfo.idl:
* Source/WebCore/Modules/WebGPU/GPUDeviceLostReason.idl:
* Source/WebCore/Modules/WebGPU/GPUErrorFilter.idl:
* Source/WebCore/Modules/WebGPU/GPUExtent3DDict.idl:
* Source/WebCore/Modules/WebGPU/GPUExternalTexture.idl:
* Source/WebCore/Modules/WebGPU/GPUExternalTextureBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUExternalTextureDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUFeatureName.idl:
* Source/WebCore/Modules/WebGPU/GPUFilterMode.idl:
* Source/WebCore/Modules/WebGPU/GPUFragmentState.idl:
* Source/WebCore/Modules/WebGPU/GPUFrontFace.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyBuffer.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyExternalImage.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyTexture.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyTextureTagged.idl:
* Source/WebCore/Modules/WebGPU/GPUImageDataLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUIndexFormat.idl:
* Source/WebCore/Modules/WebGPU/GPUInternalError.idl:
* Source/WebCore/Modules/WebGPU/GPULoadOp.idl:
* Source/WebCore/Modules/WebGPU/GPUMapMode.idl:
* Source/WebCore/Modules/WebGPU/GPUMipmapFilterMode.idl:
* Source/WebCore/Modules/WebGPU/GPUMultisampleState.idl:
* Source/WebCore/Modules/WebGPU/GPUObjectBase.idl:
* Source/WebCore/Modules/WebGPU/GPUObjectDescriptorBase.idl:
* Source/WebCore/Modules/WebGPU/GPUOrigin2DDict.idl:
* Source/WebCore/Modules/WebGPU/GPUOrigin3DDict.idl:
* Source/WebCore/Modules/WebGPU/GPUOutOfMemoryError.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineBase.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineDescriptorBase.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineError.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineErrorInit.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineErrorReason.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineLayoutDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUPowerPreference.idl:
* Source/WebCore/Modules/WebGPU/GPUPredefinedColorSpace.idl:
* Source/WebCore/Modules/WebGPU/GPUPrimitiveState.idl:
* Source/WebCore/Modules/WebGPU/GPUPrimitiveTopology.idl:
* Source/WebCore/Modules/WebGPU/GPUProgrammablePassEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPUProgrammableStage.idl:
* Source/WebCore/Modules/WebGPU/GPUQuerySet.idl:
* Source/WebCore/Modules/WebGPU/GPUQuerySetDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUQueryType.idl:
* Source/WebCore/Modules/WebGPU/GPUQueue.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundle.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundleDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoderDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURenderEncoderBase.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassColorAttachment.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassDepthStencilAttachment.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassLayout.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassTimestampLocation.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassTimestampWrite.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPipeline.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPipelineDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl:
* Source/WebCore/Modules/WebGPU/GPUSampler.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerBindingType.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderModule.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderModuleCompilationHint.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderModuleDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderStage.idl:
* Source/WebCore/Modules/WebGPU/GPUStencilFaceState.idl:
* Source/WebCore/Modules/WebGPU/GPUStencilOperation.idl:
* Source/WebCore/Modules/WebGPU/GPUStorageTextureAccess.idl:
* Source/WebCore/Modules/WebGPU/GPUStorageTextureBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUStoreOp.idl:
* Source/WebCore/Modules/WebGPU/GPUSupportedFeatures.idl:
* Source/WebCore/Modules/WebGPU/GPUSupportedLimits.idl:
* Source/WebCore/Modules/WebGPU/GPUTexture.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureAspect.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureDimension.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureFormat.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureSampleType.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureUsage.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureView.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureViewDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureViewDimension.idl:
* Source/WebCore/Modules/WebGPU/GPUUncapturedErrorEvent.idl:
* Source/WebCore/Modules/WebGPU/GPUUncapturedErrorEventInit.idl:
* Source/WebCore/Modules/WebGPU/GPUValidationError.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexAttribute.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexBufferLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexFormat.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexState.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexStepMode.idl:
* Source/WebCore/Modules/WebGPU/NavigatorGPU.idl:
* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::createContextWebGPU):
(WebCore::HTMLCanvasElement::getContextWebGPU):
* Source/WebCore/html/canvas/GPUCanvasContext.idl:
* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::gpu):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::shouldSetContentsDisplayDelegate const):

Canonical link: <a href="https://commits.webkit.org/264033@main">https://commits.webkit.org/264033@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0e17518984ae0ea83f16e00101ce8ac0af15b59d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6465 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6673 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6854 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8044 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6760 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6461 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6994 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6624 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9653 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6578 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6637 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5859 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8123 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/4102 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5839 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13689 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/5443 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/6023 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5918 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/8246 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/6056 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6401 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5235 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/6599 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5807 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1537 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1530 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9966 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/6781 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6178 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1666 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (changes requested):
This seems to cause a regression in the tests:

FAIL :serialize: assert_unreached:
 3  - EXCEPTION: WebGPU device failed to initialize with Error "No WebGPU implementation found"; not retrying
 4    assert@http://127.0.0.1:8000/webgpu/common/util/util.js:37:20

litherum committed:
[WebGPU] Rename the setting from just "WebGPU" to "WebGPUEnabled"
https://bugs.webkit.org/show_bug.cgi?id=256575
rdar://109134933

Reviewed by Simon Fraser and Mike Wyrzykowski.

This is so we can say settings().webGPUEnabled() instead of settings().webGPU(),
which is a bit more ergonomic.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/WebGPU/GPU.idl:
* Source/WebCore/Modules/WebGPU/GPUAdapter.idl:
* Source/WebCore/Modules/WebGPU/GPUAdapterInfo.idl:
* Source/WebCore/Modules/WebGPU/GPUAddressMode.idl:
* Source/WebCore/Modules/WebGPU/GPUAutoLayoutMode.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroup.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupEntry.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupLayoutDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUBindGroupLayoutEntry.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendComponent.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendFactor.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendOperation.idl:
* Source/WebCore/Modules/WebGPU/GPUBlendState.idl:
* Source/WebCore/Modules/WebGPU/GPUBuffer.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferBinding.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferBindingType.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferMapState.idl:
* Source/WebCore/Modules/WebGPU/GPUBufferUsage.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasCompositingAlphaMode.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl:
* Source/WebCore/Modules/WebGPU/GPUColorDict.idl:
* Source/WebCore/Modules/WebGPU/GPUColorTargetState.idl:
* Source/WebCore/Modules/WebGPU/GPUColorWrite.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandBuffer.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandBufferDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandEncoderDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUCommandsMixin.idl:
* Source/WebCore/Modules/WebGPU/GPUCompareFunction.idl:
* Source/WebCore/Modules/WebGPU/GPUCompilationInfo.idl:
* Source/WebCore/Modules/WebGPU/GPUCompilationMessage.idl:
* Source/WebCore/Modules/WebGPU/GPUCompilationMessageType.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassTimestampLocation.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePassTimestampWrite.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePipeline.idl:
* Source/WebCore/Modules/WebGPU/GPUComputePipelineDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUCullMode.idl:
* Source/WebCore/Modules/WebGPU/GPUDebugCommandsMixin.idl:
* Source/WebCore/Modules/WebGPU/GPUDepthStencilState.idl:
* Source/WebCore/Modules/WebGPU/GPUDevice.idl:
* Source/WebCore/Modules/WebGPU/GPUDeviceDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUDeviceLostInfo.idl:
* Source/WebCore/Modules/WebGPU/GPUDeviceLostReason.idl:
* Source/WebCore/Modules/WebGPU/GPUErrorFilter.idl:
* Source/WebCore/Modules/WebGPU/GPUExtent3DDict.idl:
* Source/WebCore/Modules/WebGPU/GPUExternalTexture.idl:
* Source/WebCore/Modules/WebGPU/GPUExternalTextureBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUExternalTextureDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUFeatureName.idl:
* Source/WebCore/Modules/WebGPU/GPUFilterMode.idl:
* Source/WebCore/Modules/WebGPU/GPUFragmentState.idl:
* Source/WebCore/Modules/WebGPU/GPUFrontFace.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyBuffer.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyExternalImage.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyTexture.idl:
* Source/WebCore/Modules/WebGPU/GPUImageCopyTextureTagged.idl:
* Source/WebCore/Modules/WebGPU/GPUImageDataLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUIndexFormat.idl:
* Source/WebCore/Modules/WebGPU/GPUInternalError.idl:
* Source/WebCore/Modules/WebGPU/GPULoadOp.idl:
* Source/WebCore/Modules/WebGPU/GPUMapMode.idl:
* Source/WebCore/Modules/WebGPU/GPUMipmapFilterMode.idl:
* Source/WebCore/Modules/WebGPU/GPUMultisampleState.idl:
* Source/WebCore/Modules/WebGPU/GPUObjectBase.idl:
* Source/WebCore/Modules/WebGPU/GPUObjectDescriptorBase.idl:
* Source/WebCore/Modules/WebGPU/GPUOrigin2DDict.idl:
* Source/WebCore/Modules/WebGPU/GPUOrigin3DDict.idl:
* Source/WebCore/Modules/WebGPU/GPUOutOfMemoryError.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineBase.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineDescriptorBase.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineError.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineErrorInit.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineErrorReason.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUPipelineLayoutDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUPowerPreference.idl:
* Source/WebCore/Modules/WebGPU/GPUPredefinedColorSpace.idl:
* Source/WebCore/Modules/WebGPU/GPUPrimitiveState.idl:
* Source/WebCore/Modules/WebGPU/GPUPrimitiveTopology.idl:
* Source/WebCore/Modules/WebGPU/GPUProgrammablePassEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPUProgrammableStage.idl:
* Source/WebCore/Modules/WebGPU/GPUQuerySet.idl:
* Source/WebCore/Modules/WebGPU/GPUQuerySetDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUQueryType.idl:
* Source/WebCore/Modules/WebGPU/GPUQueue.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundle.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundleDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoderDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURenderEncoderBase.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassColorAttachment.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassDepthStencilAttachment.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassLayout.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassTimestampLocation.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassTimestampWrite.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPipeline.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPipelineDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl:
* Source/WebCore/Modules/WebGPU/GPUSampler.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerBindingType.idl:
* Source/WebCore/Modules/WebGPU/GPUSamplerDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderModule.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderModuleCompilationHint.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderModuleDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUShaderStage.idl:
* Source/WebCore/Modules/WebGPU/GPUStencilFaceState.idl:
* Source/WebCore/Modules/WebGPU/GPUStencilOperation.idl:
* Source/WebCore/Modules/WebGPU/GPUStorageTextureAccess.idl:
* Source/WebCore/Modules/WebGPU/GPUStorageTextureBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUStoreOp.idl:
* Source/WebCore/Modules/WebGPU/GPUSupportedFeatures.idl:
* Source/WebCore/Modules/WebGPU/GPUSupportedLimits.idl:
* Source/WebCore/Modules/WebGPU/GPUTexture.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureAspect.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureBindingLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureDimension.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureFormat.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureSampleType.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureUsage.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureView.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureViewDescriptor.idl:
* Source/WebCore/Modules/WebGPU/GPUTextureViewDimension.idl:
* Source/WebCore/Modules/WebGPU/GPUUncapturedErrorEvent.idl:
* Source/WebCore/Modules/WebGPU/GPUUncapturedErrorEventInit.idl:
* Source/WebCore/Modules/WebGPU/GPUValidationError.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexAttribute.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexBufferLayout.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexFormat.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexState.idl:
* Source/WebCore/Modules/WebGPU/GPUVertexStepMode.idl:
* Source/WebCore/Modules/WebGPU/NavigatorGPU.idl:
* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::createContextWebGPU):
(WebCore::HTMLCanvasElement::getContextWebGPU):
* Source/WebCore/html/canvas/GPUCanvasContext.idl:
* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::gpu):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::shouldSetContentsDisplayDelegate const):

Canonical link: https://commits.webkit.org/264033@main
